scripts/ubuntu-bartender/ubuntu-bartender

```diff
@@ -107,9 +107,22 @@ AWS_ARM_BUILD=false
 
 # MULTIPASS SPECIFIC DEFAULTS
 MULTIPASS_DISK_SIZE=50G
-MULTIPASS_MEM_SIZE=8G
 MULTIPASS_IMAGE=daily:b # daily image of bionic
 
+# By default, do not exceed half the total memory or 8GiB (whichever is smaller).
+# All calculations are in MiB
+TOTAL_MEM=$(free -m | grep -oP '\d+' | head -n 1)
+HALF_MEM="$(( ${TOTAL_MEM} / 2 ))"
+CAP_MEM=8192
+
+if [[ "$HALF_MEM" -lt "$CAP_MEM" ]]; then
+    MULTIPASS_MEM_SIZE="${HALF_MEM}M"
+else
+    MULTIPASS_MEM_SIZE="${CAP_MEM}M"
+fi
+
+# MULTIPASS_MEM_SIZE=8G  # you can optionally override this
```

Comment:
and then here: `MULTIPASS_MEM_SIZE=${MULTIPASS_MEM_SIZE:-$MULTIPASS_MEM_SIZE_DEFAULT}`

Reply:
Done.